### PR TITLE
introduce minimal fee for transaction inclusion

### PR DESCRIPTION
**Pull request by sennui**

### CODE REVIEW

File: apps/aecore/test/aec_tx_pool_tests.erl [hunk 1]
```diff
@@ -80,7 +82,14 @@ tx_pool_test_() ->
...
        end},
-      {"ensure nonce limit for sender without account in state",
+      {"ensure mempool checks minimal miners fee",
+        fun() ->
+               PK = new_pubkey(),
+               meck:expect(aec_tx_pool, min_miners_fee, 0, 120000000000),
+               ?assertEqual(ok, aec_tx_pool:push( a_signed_tx(PK, me, 1, 20000)))
+        end
+      },
+      {"get",
```
Comment (zp-sd) [inline apps/aecore/test/aec_tx_pool_tests.erl]: get? 🤔

File: apps/aecore/src/aec_tx_pool.erl [hunk 2]
```diff
@@ -756,6 +775,13 @@ check_minimum_fee(Tx0, _Hash, _Event) ->
         false -> {error, too_low_fee}
     end.
 
+check_minimum_miner_fee(Tx0, _Hash, _Event) ->
+    Tx = aetx_sign:tx(Tx0),
+    case aetx:fee(Tx) >= aec_tx_pool:min_miners_fee() of
+        true  -> ok;
+        false -> {error, too_low_fee_for_this_miner}
```
Comment (zp-sd) [inline apps/aecore/src/aec_tx_pool.erl]: Ok, so this is a check when accepting tx into the mempool, correct?

File: apps/aecore/src/aec_tx_pool.erl [hunk 3]
```diff
@@ -263,11 +270,18 @@ init([]) ->
     ok = aec_db:ensure_transaction(fun() -> aec_db:fold_mempool(InitF, ok) end),
     ets:delete(Handled),
     lager:debug("init: GCHeight = ~p", [GCHeight]),
-    {ok, #state{gc_height = GCHeight}}.
+    MinMinersFee = case aeu_env:find_config([<<"mining">>, <<"min_miners_fee">>],
+                                                [user_config, schema_default]) of
+                                {ok, FeeVal} -> FeeVal;
+                                undefined       -> 0
```
Comment (zp-sd) [inline apps/aecore/src/aec_tx_pool.erl]: So the default to 0 in record definition is redundant, but no harm.

File: apps/aeutils/priv/aeternity_config_schema.json [hunk 4]
```diff
@@ -405,6 +405,12 @@
                     "default": 3000,
                     "minimum" : 1
                 },
+                "min_miners_fee" : {
+                    "description" : "Fee required by miner to include incoming transaction",
+                    "type" : "integer",
+                    "default": 0,
```
Comment (zp-sd) [inline apps/aeutils/priv/aeternity_config_schema.json]: `default = 0`, and `minimum = 1` doesn't make much sense? 😉

File: apps/aeutils/priv/aeternity_config_schema.json [hunk 5]
```diff
@@ -405,6 +405,12 @@
                     "default": 3000,
                     "minimum" : 1
                 },
+                "min_miners_fee" : {
+                    "description" : "Fee required by miner to include incoming transaction",
```
Comment (zp-sd) [inline apps/aeutils/priv/aeternity_config_schema.json]: I'd emphasise that that's to include transations into the mempool.

**Comment by zp-sd** [comment]
@sennui Also, this could be a PR against `minerva`?

**Comment by hanssv** [comment]
I think we want something more like #2177 - not mixing between fees and gas price?!

**Pull request closed without merging**